### PR TITLE
[docs] New code blocks

### PR DESCRIPTION
**Pull request by vladmoroz**
See https://deploy-preview-764--base-ui.netlify.app/new/components/dialog

### CODE REVIEW

File: docs/src/app/(content)/layout.tsx [hunk 1]
```diff
@@ -3,7 +3,8 @@ import * as React from 'react';
 import { AppBar } from 'docs/src/components/AppBar';
 import { Navigation } from 'docs/src/components/Navigation';
 import routes from 'docs/data/pages';
-import '../../styles/style.css';
+import 'docs/src/styles.css';
+import 'docs/src/styles/style.css';
```
Comment (vladmoroz) [inline docs/src/app/(content)/layout.tsx:7]: This is temporary just to support the current setup in place, `'docs/src/styles/style.css'` will go away eventually

**Comment by vladmoroz** [comment]
@michaldudak do you have a clue why `docs` in paths breaks deploys? https://github.com/mui/base-ui/pull/764/commits/5c2549dffcf8e178e8c70ff5fd52f8d01ac50c40

I was having minor problems with path resolution that were fixed by adding `docs` path to tsconfig, but Netlify builds break then

**Comment by michaldudak** [comment]
tsconfig fields aren't merged with parent config settings but replaced, so by adding `compilerOptions.paths`, you overwrote the paths defined in `/tsconfig.base.json`. Try adding the path to the base config instead.

**Comment by michaldudak** [comment]
The demos on old pages are getting messed up a bit. Would it be problematic to actually edit the styles on the existing pages instead of the `/new` ones?

File: docs/src/styles.css [hunk 2]
```diff
@@ -1,46 +1,52 @@
...
-@import 'tailwindcss/base';
-@import 'tailwindcss/components';
-@import 'tailwindcss/utilities';
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
 
 :root {
-  --color-content: #fff;
+  --font-mono: 'Menlo', 'Consolas', monospace;
```
Comment (michaldudak) [inline docs/src/styles.css:11]: Is the default font OK for Linux users?

File: docs/src/components/demo/loadDemo.ts [hunk 3]
```diff
@@ -127,9 +126,9 @@ async function loadSimpleDemo(path: string, variantName: string): Promise<DemoVa
   const jsFilePath = mainFilePath.replace(/\.tsx?$/, '.js');
   if (mainFileLanguage === 'ts' && existsSync(jsFilePath)) {
     const jsContent = await readFile(jsFilePath, 'utf-8');
-    const jsPrettyPromise = await codeToHtml(jsContent, {
+    const jsPrettyPromise = highlighter.codeToHtml(jsContent, {
```
Comment (michaldudak) [inline docs/src/components/demo/loadDemo.ts:129]:
```suggestion
    const jsPrettyContent = highlighter.codeToHtml(jsContent, {
```

File: docs/src/styles.css (hunk 2 above)
Comment (vladmoroz) [reply docs/src/styles.css:11]: Good point, I'll add a couple more fallbacks

**Comment by vladmoroz** [comment]
> The demos on old pages are getting messed up a bit. Would it be problematic to actually edit the styles on the existing pages instead of the `/new` ones?

Oh man this annoys me the hell out of me, but I don't know if it's worth our time to refactor styles on the old pages. The root issue is that `.Content ...` selector that targets all the elements on the page indiscriminately and messes up styles of anything you put on the page. I think we'll have to live with it for a bit, because otherwise I just need to refactor a thing we won't use soon at all

Anything particular that's messed up and bothers you?

**Comment by vladmoroz** [comment]
@michaldudak looks like unrelated tests are failing after rebasing, I assume I can ignore them?

**Comment by michaldudak** [comment]
>  I think we'll have to live with it for a bit, because otherwise I just need to refactor a thing we won't use soon at all
> 
> Anything particular that's messed up and bothers you?

It's nothing blocking. The main issue is that demos render in dark mode (following my `prefers-color-scheme`), while the rest of the page is light, and they have some extra horizontal scrollbars.

---

Aside from this, I just realized it could be nice to detect cmd/ctrl + a when a demo source is focused and select just the demo source instead of the whole page (could be done in a separate PR, of course)

**Comment by michaldudak** [comment]
As for the broken tests - this looks like an issue after migrating to vitest. I'll see what's going on right away.

EDIT: likely fixed in https://github.com/mui/base-ui/pull/765

**Comment by vladmoroz** [comment]
> Aside from this, I just realized it could be nice to detect cmd/ctrl + a when a demo source is focused and select just the demo source instead of the whole page (could be done in a separate PR, of course)

Agree, added that

> As for the broken tests - this looks like an issue after migrating to vitest. I'll see what's going on right away.
>
> EDIT: likely fixed in https://github.com/mui/base-ui/pull/765

Thanks!

**Review by colmtuite — APPROVED** [review]

**Comment by michaldudak** [comment]
One thing I'm missing is the Reset button. It used to come in handy to reload the demo to its initial state.

**Comment by vladmoroz** [comment]
> One thing I'm missing is the Reset button. It used to come in handy to reload the demo to its initial state.

We decided to remove it, among other things, since there was too much going on in the toolbar. I agree it has its uses, but it's probably more so for us as we maintain and test the components in the docs way more often than the end user would.

**Pull request merged**